Makefile

```diff
@@ -37,13 +39,22 @@ test-unit:
 # Run the code generation tasks.
 # Example:
 #    make update
-update: install-go-bindata
+update: install-go-bindata install-go-deps
 	hack/update-codegen.sh
 	hack/update-generated-bindata.sh
 
 install-go-bindata:
 	hack/install-go-bindata.sh
 
+install-go-deps:
+	go mod tidy
+	go mod vendor
+	go mod verify
+	# go mod does not vendor in scripts so we need to get them manually
+	# NOTE: Make sure to keep this up-to-date with the version specified in go.mod
```

Comment:
I really hate this - I know it's go mod fault (there are issues in Go's github describing why they're not letting ppl do this). This is highly sub optimal and can lead to silly skews - let's see if we can find a better solution as well (or check if someone else fixes this issue somehow)

Reply:
maybe split this and the golangci-lint mod vendoring via tools.go out into new new follow up issues and get this merged for now (also especially now that i's all green lit on CI)? May not get a quick fix here otherwise because I'm travelling atm. I havent found a good explanation of when scripts aren't vendored, maybe we can file this upstream so we get a mod pkg we can vendor that actually contains the scripts as well?

Reply:
also feel free to push follow up fix commits directly to this branch :)